lottery/src/lib.rs

```diff
@@ -0,0 +1,227 @@
+                msg::send_bytes(win_player.player_id, b"Winner", exec::value_available());
+                self.lottery_history
+                    .insert(self.lottery_id, win_player.player_id);
+                msg::reply(Event::Winner(index), 0);
+            }
+
+            debug!("Winner: {}", index);
+
+            self.lottery_state = LotteryState::default();
+            self.players = BTreeMap::new();
+            self.players_timestamp = BTreeMap::new();
+        } else {
+            debug!(
+                "pick_winner() failed! {} {}",
+                self.lottery_state.lottery_start_time + self.lottery_state.lottery_duration,
+                exec::block_timestamp()
+            );
+        }
+    }
+}
+
+static mut LOTTERY: Option<Lottery> = None;
+
+#[no_mangle]
+pub unsafe extern "C" fn handle() {
+    let action: Action = msg::load().expect("Could not load Action");
+    let lottery: &mut Lottery = LOTTERY.get_or_insert(Lottery::default());
+
+    match action {
+        Action::Enter => {
+            lottery.enter();
+        }
+
+        Action::StartLottery(duration) => {
+            lottery.start_lottery(duration);
+        }
+
+        Action::LotteryState => {
+            msg::reply(Event::LotteryState(lottery.lottery_state), 0);
+            debug!("LotteryState: {:?}", lottery.lottery_state);
+        }
+
+        Action::PickWinner => {
+            lottery.pick_winner();
+        }
+
+        Action::BalanceOf(index) => {
+            lottery.get_balance(index);
+        }
+
+        Action::GetPlayers => {
+            lottery.get_players();
+        }
+
+        Action::LeaveLottery(index) => {
+            lottery.leave_lottery(index);
+        }
+
+        Action::AddValue(index) => {
+            lottery.add_value(index);
+        }
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn init() {
+    let lottery = Lottery {
+        lottery_owner: msg::source(),
+        ..Default::default()
+    };
+
+    LOTTERY = Some(lottery);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn meta_state() -> *mut [i32; 2] {
+    let query: State = msg::load().expect("failed to decode input argument");
+    let lottery: &mut Lottery = LOTTERY.get_or_insert(Lottery::default());
+
+    let encoded = match query {
+        State::GetPlayers => StateReply::Players(lottery.players.clone()).encode(),
+        State::GetWinners => StateReply::Winners(lottery.lottery_history.clone()).encode(),
+
+        State::BalanceOf(index) => {
+            if let Some(player) = lottery.players.get(&index) {
+                StateReply::Balance(player.balance).encode()
+            } else {
+                StateReply::Balance(0).encode()
+            }
+        }
+    };
+
+    gstd::util::to_leak_ptr(encoded)
+}
+
+gstd::metadata! {
+    title: "Lottery",
+        init:
+
+        handle:
+            input: Action,
+            output: Event,
+        state:
+            input: State,
+            output: StateReply,
```

Comment:
No need for an empty entry. And indents again.
```suggestion
    handle:
        input: Action,
        output: Event,
    state:
        input: State,
        output: StateReply,
```